codegen-server/python/src/main/kotlin/software/amazon/smithy/rust/codegen/server/python/smithy/generators/PythonServerUnionGenerator.kt

```diff
@@ -0,0 +1,202 @@
+import software.amazon.smithy.rust.codegen.core.smithy.generators.UnionGenerator
+import software.amazon.smithy.rust.codegen.core.smithy.rustType
+import software.amazon.smithy.rust.codegen.core.util.isTargetUnit
+import software.amazon.smithy.rust.codegen.core.util.toSnakeCase
+import software.amazon.smithy.rust.codegen.server.python.smithy.PythonServerCargoDependency
+import software.amazon.smithy.rust.codegen.server.python.smithy.pythonType
+import software.amazon.smithy.rust.codegen.server.python.smithy.renderAsDocstring
+
+private fun RustType.renderPy() {
+}
+
+class PythonServerUnionGenerator(
+    model: Model,
+    private val symbolProvider: SymbolProvider,
+    private val writer: RustWriter,
+    shape: UnionShape,
+    private val renderUnknownVariant: Boolean = true,
+) : UnionGenerator(model, symbolProvider, writer, shape, renderUnknownVariant) {
+    private val sortedMembers: List<MemberShape> = shape.allMembers.values.sortedBy { symbolProvider.toMemberName(it) }
+    private val unionSymbol = symbolProvider.toSymbol(shape)
+
+    private val pyo3 = PythonServerCargoDependency.PyO3.toType()
+
+    override fun render() {
+        super.render()
+        renderPyUnionStruct()
+        renderPyUnionImpl()
+        renderPyObjectConverters()
+    }
+
+    private fun renderPyUnionStruct() {
+        writer.rust("""##[pyo3::pyclass(name = "${unionSymbol.name}")]""")
+        val containerMeta = unionSymbol.expectRustMetadata()
+        containerMeta.render(writer)
+        writer.rust("struct PyUnionMarker${unionSymbol.name}(pub ${unionSymbol.name});")
+    }
+
+    private fun renderPyUnionImpl() {
+        Attribute(pyo3.resolve("pymethods")).render(writer)
+        writer.rustBlock("impl PyUnionMarker${unionSymbol.name}") {
+            sortedMembers.forEach { member ->
+                val funcNamePart = member.memberName.toSnakeCase()
+                val variantName = symbolProvider.toMemberName(member)
+
+                if (sortedMembers.size == 1) {
+                    Attribute.AllowIrrefutableLetPatterns.render(this)
+                }
+                renderNewVariant(writer, model, symbolProvider, member, variantName, funcNamePart, unionSymbol)
+                renderAsVariant(writer, model, symbolProvider, member, variantName, funcNamePart, unionSymbol)
+                rust("/// Returns true if this is a [`$variantName`](#T::$variantName).", unionSymbol)
+                rust("/// :rtype bool:")
+                rustBlock("pub fn is_$funcNamePart(&self) -> bool") {
+                    rust("self.0.is_$funcNamePart()")
+                }
+            }
+            if (renderUnknownVariant) {
+                rust("/// Returns true if the union instance is the `Unknown` variant.")
+                rust("/// :rtype bool:")
+                rustBlock("pub fn is_unknown(&self) -> bool") {
+                    rust("matches!(self.0, Self::Unknown)")
```

Comment:
I think this should be `self.0.is_unknown()` because the wrapper type won't have any `Unknown` variant

Reply:
Good catch.